test/functional/ui/float_spec.lua

```diff
@@ -2034,6 +2034,61 @@ describe('float window', function()
       eq('center', footer_pos)
     end)
 
+    it('center aligned title longer than window width #25746', function()
+      local buf = meths.create_buf(false, false)
+      meths.buf_set_lines(buf, 0, -1, true, {' halloj! ',
+                                             ' BORDAA  '})
+      local win = meths.open_win(buf, false, {
+        relative='editor', width=9, height=2, row=2, col=5, border="double",
+        title = "abcdefghijklmnopqrstuvwxyz",title_pos = "center",
+      })
+
+      if multigrid then
+        screen:expect{grid=[[
+        ## grid 1
+          [2:----------------------------------------]|
+          [2:----------------------------------------]|
+          [2:----------------------------------------]|
+          [2:----------------------------------------]|
+          [2:----------------------------------------]|
+          [2:----------------------------------------]|
+          [3:----------------------------------------]|
+        ## grid 2
+          ^                                        |
+          {0:~                                       }|
+          {0:~                                       }|
+          {0:~                                       }|
+          {0:~                                       }|
+          {0:~                                       }|
+        ## grid 3
+                                                  |
+        ## grid 4
+          {5:╔}{11:abcdefghi}{5:╗}|
```

Comment:
This preserves the first half. truncate the title before render it also fixes it. like the statusline `<text` keep the second half.  not sure which is good way :thinking:

Reply:
another way is to skip text while column < 1 - truncates center aligned text from both sides (adapted from `grid_line_puts`):
```diff
diff --git a/src/nvim/drawscreen.c b/src/nvim/drawscreen.c
index adbe0f473..896d0c4a6 100644
--- a/src/nvim/drawscreen.c
+++ b/src/nvim/drawscreen.c
@@ -725,7 +725,17 @@ static void win_redr_bordertext(win_T *wp, VirtText vt, int col)
     if (text == NULL) {
       break;
     }
-    col += grid_line_puts(col, text, -1, attr);
+    while (col < 1 && *text != NUL) {
+      int u8cc[MAX_MCO];
+      int mbyte_blen = utfc_ptr2len(text);
+      int u8c = utfc_ptr2char(text, u8cc);
+      int mbyte_cells = utf_char2cells(u8c);
+      col += MIN(mbyte_cells, 2);
+      text += mbyte_blen;
+    }
+    if (col >= 1) {
+      col += grid_line_puts(col, text, -1, attr);
+    }
   }
 }
```

Reply:
Yes. I prefer to truncate the title. But I also confused. Can't the titile length and window width be calculated locally before setting them.  But in any case should avoid crash in core.

Reply:
The title length is already calculated, its just the alignment makes the text start outside the window. The truncated title needs to be calculated every time title, title_pos, and width change.